apm-agent-plugins/apm-es-restclient-plugin/apm-es-restclient-plugin-common/src/test/java/co/elastic/apm/agent/es/restclient/AbstractEsClientInstrumentationTest.java

```diff
@@ -0,0 +1,187 @@
+        assertThat(http).isNotNull();
+        assertThat(http.getMethod()).isEqualTo(method);
+        assertThat(http.getStatusCode()).isEqualTo(statusCode);
+        assertThat(http.getUrl()).isEqualTo("http://" + container.getHttpHostAddress());
+    }
+
+    protected void validateSpanContentAfterIndexCreateRequest() {
+        System.out.println(reporter.generateTransactionPayloadJson());
+
+        List<Span> spans = reporter.getSpans();
+        assertThat(spans).hasSize(1);
+        validateSpanContent(spans.get(0), String.format("Elasticsearch: PUT /%s", SECOND_INDEX), 200, "PUT");
+    }
+
+    protected void validateSpanContentAfterIndexDeleteRequest() {
+        System.out.println(reporter.generateTransactionPayloadJson());
+
+        List<Span>spans = reporter.getSpans();
+        assertThat(spans).hasSize(1);
+        validateSpanContent(spans.get(0), String.format("Elasticsearch: DELETE /%s", SECOND_INDEX), 200, "DELETE");
+    }
+
+    protected void validateSpanContentAfterBulkRequest() {
+        System.out.println(reporter.generateTransactionPayloadJson());
+
+        List<Span> spans = reporter.getSpans();
+        assertThat(spans).hasSize(1);
+        assertThat(spans.get(0).getName().toString()).isEqualTo("Elasticsearch: POST /_bulk");
+    }
+
+    protected Response doPerformRequest(String method, String path) throws IOException, ExecutionException {
+        if (async) {
+            final CompletableFuture<Response> resultFuture = new CompletableFuture<>();
+            lowLevelClient.performRequestAsync(method, path, new ResponseListener() {
+                @Override
+                public void onSuccess(Response response) {
+                    resultFuture.complete(response);
+                }
+
+                @Override
+                public void onFailure(Exception exception) {
+                    resultFuture.completeExceptionally(exception);
+                }
+            });
+            try {
+                return resultFuture.get();
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+        }
+        return lowLevelClient.performRequest(method, path);
+    }
+
+//    private void validateHttpContextContent(Http http, int statusCode, String method, String url) {
+//        assertThat(http).isNotNull();
+//        assertThat(http.getMethod()).isEqualTo(method);
+//        assertThat(http.getStatusCode()).isEqualTo(statusCode);
+//        assertThat(http.getUrl()).isEqualTo(url);
+//    }
+//
+//    private void validateDbContextContent(Span span, String statement) {
+//        Db db = span.getContext().getDb();
+//        assertThat(db.getType()).isEqualTo(ELASTICSEARCH);
+//        assertThat((Object) db.getStatementBuffer()).isNotNull();
+//        assertThat(db.getStatementBuffer().toString()).isEqualTo(statement);
+//    }
+
+
+    public abstract void testCreateAndDeleteIndex() throws IOException, ExecutionException, InterruptedException;
```

Comment:
Since all actual test implementations must declare `@Test` in any case, I think there is no need for these abstract declaration of test methods.

Reply:
deleted